client-state-machine.go

```diff
@@ -607,6 +641,14 @@ func (state clientStateWaitEE) Next(hr handshakeMessageReader) (HandshakeState,
 		state.Params.NextProto = serverALPN.Protocols[0]
 	}
 
+	if foundExts[ExtensionTypeClientCertType] &&
+		foundExts[ExtensionTypeServerCertType] &&
+		state.Config.AllowRawPublicKeys {
+		foundClient := serverClientCertType.CertificateTypes[0] == CertificateTypeRawPublicKey
+		foundServer := serverServerCertType.CertificateTypes[0] == CertificateTypeRawPublicKey
+		state.Params.UsingRawPublicKeys = foundClient && foundServer
```

Comment:
BUG: what happens if the server doesn't want client auth at all but wants to use a raw public key?